hydromt_wflow/wflow.py

```diff
@@ -2143,6 +2132,259 @@ def setup_ksathorfrac(
+        target_conductivity: list
+            List of target vertical conductivities [mm/day] for each layer in
+            ``wflow_thicknesslayers``. Set value to `None` if no specific value is
+            required, by default [None, None, 5, None, None].
+        lulc_vars : list
+            List of landuse parameters to prepare.
+            By default ["landuse","Kext","N","PathFrac","RootingDepth","Sl","Swood",
+            "WaterFrac"]
+        paddy_waterlevels : dict
+            Dictionary with the minimum, optimal and maximum water levels for paddy
+            fields [mm]. By default {"h_min": 20, "h_opt": 50, "h_max": 80}
+        save_high_resolution_lulc : bool
+            Save the high resolution landuse map merged with the paddies to the static
+            folder. By default False.
+        """
+        self.logger.info("Preparing LULC parameter maps including paddies.")
+        # Check if soil data is available
+        if "KsatVer" not in self.grid.data_vars:
+            raise ValueError(
+                "KsatVer and f are required to update the soil parameters with paddies."
+                "Please run setup_soilmaps first."
+            )
+
+        if lulc_mapping_fn is None:
+            lulc_mapping_fn = f"{lulc_fn}_mapping_default"
+        # read landuse map and mapping table
+        landuse = self.data_catalog.get_rasterdataset(
+            lulc_fn, geom=self.region, buffer=2, variables=["landuse"]
+        )
+        df_mapping = self.data_catalog.get_dataframe(
+            lulc_mapping_fn,
+            driver_kwargs={"index_col": 0},  # only used if fn_map is a file path
+        )
+        output_paddy_class = (
+            paddy_class if output_paddy_class is None else output_paddy_class
+        )
+
+        # if needed, add paddies to landuse
+        if paddy_fn is not None:
+            # Read paddy map and mapping table
+            paddy = self.data_catalog.get_rasterdataset(
+                paddy_fn, geom=self.region, buffer=2, variables=["paddy"]
+            )
+            if paddy_mapping_fn is None:
+                paddy_mapping_fn = "paddy_mapping_default"
+            df_paddy_mapping = self.data_catalog.get_dataframe(
+                paddy_mapping_fn,
+                driver_kwargs={"index_col": 0},
+            )
+
+            landuse, df_mapping = workflows.add_paddy_to_landuse(
+                landuse,
+                paddy,
+                paddy_class,
+                output_paddy_class=output_paddy_class,
+                df_mapping=df_mapping,
+                df_paddy_mapping=df_paddy_mapping,
+            )
+
+            if save_high_resolution_lulc:
+                output_dir = join(self.root, "maps")
+                if not os.path.exists(output_dir):
+                    os.makedirs(output_dir)
+                landuse.raster.to_raster(join(output_dir, "landuse_with_paddy.tif"))
+                df_mapping.to_csv(join(output_dir, "landuse_with_paddy_mapping.csv"))
```

Comment:
Nice :D

Reply:
We needed it for our project already to do scenarios :)